examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,104 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+#include <ArborX_InterpMovingLeastSquares.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+
+using Point = ArborX::ExperimentalHyperGeometry::Point<2, double>;
+
+KOKKOS_FUNCTION double functionToApproximate(Point const &p)
+{
+  return Kokkos::cos(p[0] + p[1] / 4);
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = typename ExecutionSpace::memory_space;
+  ExecutionSpace space{};
+
+  // Source space is a 3x3 grid
+  // Target space is a single point
+  //
+  //  ^
+  //  |
+  //  S     S     S
+  //  |
+  //  |   T
+  //  S     S  T  S
+  //  |
+  //  | T
+  // -S-----S-----S->
+  //  |
+
+  // Set up points
+  Kokkos::View<Point *, MemorySpace> src_points("Example::src_points", 9);
+  Kokkos::View<Point *, MemorySpace> tgt_points("Example::tgt_points", 3);
```

Comment:
```suggestion
  int const num_sources = 9;
  int const num_targets = 3;
  Kokkos::View<Point *, MemorySpace> src_points("Example::src_points", num_sources);
  Kokkos::View<Point *, MemorySpace> tgt_points("Example::tgt_points", num_targets);
```
and eliminate all hardcoded `3`s and `9`s below, using variables instead.